bot/cogs/reddit.py

```diff
@@ -34,6 +41,66 @@ def __init__(self, bot: Bot):
         self.new_posts_task = None
         self.top_weekly_posts_task = None
 
+        self.refresh_access_token.start()
+
+    @tasks.loop(hours=0.99)  # access tokens are valid for one hour
+    async def refresh_access_token(self) -> None:
+        """Refresh the access token"""
+        headers = {"Authorization": self.client_auth}
+        data = {
+            "grant_type": "refresh_token",
+            "refresh_token": self.refresh_token
+        }
+
+        response = await self.bot.http_session.post(
+            url = f"{self.URL}/api/v1/access_token",
+            headers=headers,
+            data=data,
+        )
+
+        content = await response.json()
+        self.access_token = content["access_token"]
+        self.headers = {
+            "Authorization": "bearer " + self.access_token,
+            "User-Agent": self.USER_AGENT
+        }
+
+    @refresh_access_token.before_loop
+    async def get_tokens(self) -> None:
+        """Get Reddit access and refresh tokens"""
+        await self.bot.wait_until_ready()
+
+        headers = {"User-Agent": self.USER_AGENT}
+        data = {
+            "grant_type": "client_credentials",
+            "duration": "permanent"
+        }
+
+        if RedditConfig.client_id and RedditConfig.secret:
+            self.client_auth = BasicAuth(RedditConfig.client_id, RedditConfig.secret)
+
+            response = await self.bot.http_session.post(
+                url=f"{self.URL}/api/v1/access_token",
+                headers=headers,
+                auth=self.client_auth,
+                data=data
+            )
+
+            content = await response.json()
+            self.access_token = content["access_token"]
+            self.refresh_token = content["refresh_token"]
+            self.headers = {
+                "Authorization": "bearer " + self.access_token,
+                "User-Agent": self.USER_AGENT
+            }
+        else:
+            self.client_auth = None
+            self.access_token = None
+            self.refresh_token = None
+            self.headers = None
+
+            log.error("Unable to find client credentials.")
```

Comment:
What's the actual implication of this? The cog doesn't work?

Reply:
Yes pretty much. Without client id or secret, we can not get an acces token and therefore not use Reddit API. 

Perhaps this should be changed to something more descriptive like `No client credentials were provided. Cannot use Reddit API` and unload the cog?

Reply:
Yeah, that sounds like a good idea.

Reply:
It would be even more robust to check if authentication succeeds rather than just if the credentials are present. In any case, I agree that unloading the extension should be done.